server/doc.js

```diff
@@ -0,0 +1,36 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+
+const DocSchema = new Schema({
+  title: {type: String, required:true},
+  content: {type: String, required:true},
+  owner_id: {type: Schema.Types.ObjectId, ref: 'User'},
+  locked: Boolean,
+  revisions:[
+    {type: Schema.Types.ObjectId, ref: 'Revision'}
+  ],
+  date_created: {type: Date, 
+    default: Data.now
+  }
+})
+
+DocSchema.methods.findAllTabooIdx = function(taboo_words){
+  //A String is considered a word if there is a space
+  //Checks all the words and see if any are in taboo words and replaces those words
+  //returns indexes of all taboo words, format at each index: [start, end + 1]
+  idx_list = [];
+//  for(let i = 0; i < taboo_words.length; i++){
+  for (let taboo_word in taboo_words){
```

Comment:
Using "in" is incorrect.
This should be (const taboo_word of taboo_words).

Reply:
changed